src/compute/src/render/continual_task.rs

```diff
@@ -393,11 +393,22 @@ fn continual_task_sink<G: Scope<Timestamp = Timestamp>>(
             debug!("ct_sink got write {:?}: {:?}", new_upper, to_append);
 
             let mut expected_upper = write_handle.shared_upper();
-            while PartialOrder::less_than(&expected_upper, &new_upper) {
+            loop {
+                if !PartialOrder::less_than(&expected_upper, &new_upper) {
+                    state.output_progress = expected_upper.clone();
+                    debug!("ct_sink skipping {:?}", new_upper.elements());
+                    break;
+                }
```

Comment:
In `state.process()` we already skip data that is behind `state.output_progress` so mechanically it seems like skipping the `to_append` batch here should be okay!